website/docs/docs/build/udfs.md

```diff
@@ -0,0 +1,208 @@
+
+dbt creates, updates, and renames UDFs as part of DAG execution. The UDF file is created before building the model that references it.
+
+## Supported adapters
+
+UDFs are supported in the following adapters:
+
+<Tabs>
+
+<TabItem value="core" label="dbt Core local install">
+
+- BigQuery
+- Snowflake
+- Redshift
+- Postgres
+- Databricks
+
+</TabItem>
+
+<TabItem value ="dbt platform" label ="dbt Core in the dbt platform">
+
+- BigQuery
+- Snowflake
+- Redshift
+- Databricks
+
+</TabItem>
+</Tabs>
+
+## Defining UDFs in dbt
+
+To define UDFs in dbt, refer to the following steps:
+
+1. Create a SQL file under the `functions` directory. For example:
+
+    <File name='functions/is_positive_int.sql'>
+
+    ```sql
+    {{ config(
+        database='udf_db'
+    ) }}
+
+    REGEXP_CONTAINS(a_string, r'^[0-9]+$')
+    ```
+
+    </File>
+
+2. Specify the function name and define the config, properties, return type, and optional arguments in a corresponding YAML file. For example:
+
+
+    <File name='functions/schema.yml'>
+
+    ```yml
+    functions:
+      - name: is_positive_int # required
+        description: My UDF that determines if a string represents a positive (+) integer # required
+        config:
+            schema: udf_schema
+            database: udf_db
+        arguments: # optional
+            - name: a_string # required if arguments is specified
+              data_type: string # required if arguments is specified
+              description: The string that I want to check if it's representing a positive integer (like "10") 
+        returns: # required
+            data_type: boolean # required
+    ```
+    </File>
+
+    <!--other types not yet supported
+    <Expandable alt_header="Supported UDF types">
+
+    You can use these values for the `type` property when you define a function in a YAML file.
+
+    - `scalar` - Returns a single value per row
+    - `aggregate` - Returns a single value per group, aggregating several rows
+    - `table functions` - Returns a table result
+    <br></br>
+    For example:
+
+    ```yml
+    functions:
+	  - name: string 
+	    description: string
+	    type: scalar # default value
+    ```
+
+    If not explicitly specified, the `type` property defaults to `scalar`.
+
+    </Expandable>
+    -->
+    
+    The rendered `CREATE` UDF statement depends on which adapter you’re using. For example:
+
+    <Tabs>
+
+    <TabItem value="Snowflake">
+    ```sql
+    CREATE OR REPLACE FUNCTION my_schema.is_positive_int(a_string STRING)
+    RETURNS BOOLEAN
+    AS (
+    REGEXP_CONTAINS(a_string, r'^[0-9]+$')
+    );
+    ```
+    </TabItem>
+
+    <TabItem value="Redshift">
+    ```sql
+    CREATE OR REPLACE FUNCTION repeat_n(s VARCHAR, n INTEGER)
+    RETURNS VARCHAR
+    VOLATILE
+    AS $$ SELECT repeat(s, n); $$
+    LANGUAGE SQL;
```

Comment:
to take a guess at syntax looking at redshift [docs](https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_FUNCTION.html) (i HAVE NOT VALIDATED)

```suggestion
    CREATE OR REPLACE FUNCTION udf_db.udf_schema.is_positive_int(a_string STRING)
    RETURNS BOOLEAN
    STABLE
    AS $$ SELECT REGEXP_CONTAINS(a_string, r'^[0-9]+$'); $$
    LANGUAGE SQL;
```

things i don't know:
- does redshift adapter automatically inject the `SELECT` or do you need to add that to the .sql file?
- i think volatility is stable for this example?

Reply:
Volatility isn't set-able yet. It's hard coded in redshift because we have to supply a volitility redshift (it doesn't let you not)